Tests/SonarScanner.MSBuild.Tasks.IntegrationTests/TargetsTests/ImportBeforeTargetsTests.cs

```diff
@@ -54,17 +54,17 @@ public void ImportsBefore_SonarQubeTargetsPathNotSet()
         {
             // 1. Prebuild
             // Arrange
-            EnsureDummyIntegrationTargetsFileExists();
-
-            var preImportProperties = new WellKnownProjectProperties
-            {
-                SonarQubeTargetsPath = "",
-                TeamBuild2105BuildDirectory = "",
-                TeamBuildLegacyBuildDirectory = ""
-            };
+            var projectXml = @"
+<PropertyGroup>
+  <SonarQubeTargetsPath />
+  <AGENT_BUILDDIRECTORY />
+  <TF_BUILD_BUILDDIRECTORY />
+</PropertyGroup>
+";
+            var projectFilePath = CreateProjectFile(projectXml);
 
             // Act
-            var projectInstance = CreateAndEvaluateProject(preImportProperties);
+            var projectInstance = CreateAndEvaluateProject(projectFilePath);
```

Comment:
I've refactor five test classes. Each test class was producing a slightly different type of project file (i.e. more or less properties set by default), depending on what is required by the target(s) being tested.

For each test class, I looked at the format of the basic project file that was produced and created an XML template for that class i.e. the templates are different in each case.
I ran the old and new versions of the projects side-by-side and diff-ed the project file that were generated to make sure they are equivalent.